packages/nimble-components/src/spinner/tests/spinner.stories.ts

```diff
@@ -3,12 +3,39 @@ import type { Meta, StoryObj } from '@storybook/html';
 import { withXD } from 'storybook-addon-xd-designs';
 import { createUserSelectedThemeStory } from '../../utilities/tests/storybook';
 import '../../all-components';
+import {
+    spinnerLargeHeight,
+    spinnerMediumHeight
+} from '../../theme-provider/design-tokens';
 
-const overviewText = 'The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent loading progress, or an ongoing operation, of an indeterminate / unknown duration.'
-    + '<p>The default spinner size is 16x16. Other sizes can be set via width/height CSS styles on the component, and it will scale appropriately.</p>'
-    + '<p>Other sizes suggested by the Nimble designers: 32x32, 48x48, 64x64, 96x96, 128x128.</p>';
+const spinnerSize = {
+    small: null,
+    medium: `height: var(${spinnerMediumHeight.cssCustomProperty});`,
+    large: `height: var(${spinnerLargeHeight.cssCustomProperty});`
+} as const;
 
-const metadata: Meta = {
+interface SpinnerArgs {
+    size: keyof typeof spinnerSize;
+}
+
+const overviewText = '<p>The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent '
+    + 'loading progress, or an ongoing operation, of an indeterminate / unknown duration.</p>'
+    + '<p>To customize its size, set its CSS `height` to a design token, and its width will automatically match its height.</p>'
+    + '<ul>'
+    + '<li>16x16 (small, default)'
```

Comment:
Right now we're inconsistent about whether the primary API is "small/medium/large" or "16/32/64". In this file and the matrix story the pixel sizes are prominent and the t-shirt sizes are extra information, but in the design token documentation it's the opposite. I'm glad we describe both in most places, just think we should be consistent about which one is our "official" terminology.

I'd vote for small/medium/large to be the "official" naming, which would mean changing a few places in this file and the matrix story to be more like "medium (32x32)".

Reply:
Updated